### ISSUE TITLE
Tuple with unpacked TypeVarTuple Param has incorrectly unioned last element

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#8485]
Take this example:

```python
from typing import reveal_type

class Container[*Ts]:
    def foo(self) -> tuple[*Ts, str]: ...

container: Container[*tuple[int, ...]] = Container()
last_elem = container.foo()
reveal_type(last_elem[-1]) # revealed type is `str | int`
```

[Pyright Playground
](https://pyright-play.net/?pyrightVersion=1.1.350&enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMogCmAboQIYA2A%2BvAoQFCMDGFZAzm1AMJgoxmpCIANoAqACpsAugC56UBVAAmhYFGBgwACjaEKwAJRQAtAD5YAVwQVCYyQBoobGCFlQAdJ%2Ba9%2BgkDO4fARQhMRgrG2FUGEdPdykpKABeQL5goS0DelZnKj1CaBSmIL93DW0solJKGkRCLRyYPJsIYWMARikDIA)

When using ellipses to express a variable amount of tuple elements, for the unpacked expression, Pyright unions the last element in the tuple with the proceeding element before the ellipses, MyPy seems to handle this case correctly: [MyPy Playground](https://mypy-play.net/?mypy=latest&python=3.12&enable-incomplete-feature=TypeVarTuple&gist=1690d7258172e4a0e535c20407cc3d19)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.373.
